shared.go

```diff
@@ -0,0 +1,25 @@
+package subscription
+
+import "time"
+
+type SubscriptionInfo struct {
```

Comment:
No need here, but usually for SDK quality code we require all exported Go types/functions/vars to have Godoc